editor/dependency_editor.cpp

```diff
@@ -578,32 +578,41 @@ void DependencyRemoveDialog::ok_pressed() {
 		}
 	}
 
+	bool project_settings_modified = false;
```

Comment:
I would refactor a bit to avoid redundancy:
```cpp
bool project_settings_modified = true;
for (int i = 0; i < files_to_delete.size(); ++i) {
	// If the file we are deleting for e.g. the main scene, default environment,
	// or audio bus layout, we must clear its definition in Project Settings.
	const String &file = files_to_delete[i];
	if (file == String(GLOBAL_GET("application/config/icon"))) {
		ProjectSettings::get_singleton()->set("application/config/icon", "");
	} else if (file == String(GLOBAL_GET("application/run/main_scene"))) {
		ProjectSettings::get_singleton()->set("application/run/main_scene", "");
	} else if (  ...
... etc.
	} else {
		project_settings_modified = false;
	}
```

Reply:
I don't think setting the bool to false only when no match is found fixes this since this method is being called anytime a file is removed not just one that has ties to project settings. This would cause issue when the user is deleting multiple files and one of them is not related to any of these fields depending on the order of how the file is processed it would cause the same issue.

I have also tested and confirmed this approach produce the same bug when deleting multi files.

Reply:
I'm pretty sure the approach I suggest is strictly equivalent to the current code.
- Current code is: Assume false, if there's a match, set to true.
- My suggestion: Assume true, if there's no match, set to false.

This should lead to the same state at the end of the big `if`/`else if`/`else` block, as there can be only two options:
- Either there's a match, and it will stay `true`
- Or there's no match, it enters the `else`, and becomes `false`

Reply:
Well there's one case where my approach would lead to different behavior, which is if the `for` loop never runs, i.e. `files_to_delete.size() == 0`.

I don't know if this can happen, but that's right, let's keep the defensive form then.

Reply:
The issue happens if we assume this value for `files_to_delete` with your approach
["the_main_scene", "completely_unrelated_file"]

The first iteration will leave the flag at `true`, and then the 2nd iteration will set it to `false` and then no actual save will happen since I only save after loop has finished and the flag is final.
https://github.com/godotengine/godot/blob/138bcc7cfdbddcc05a4f2373fb2cb20fb23e0f2b/editor/dependency_editor.cpp#L617-L622

Reply:
You're totally right. Thanks for bearing with me :D

Reply:
Cheers! 🍺